CONTRIBUTING.md

```diff
@@ -1,18 +1,149 @@
-# Contributing
+# Contributing to BinderHub
 
-Welcome! As a [Jupyter](https://jupyter.org) project, we follow the [Jupyter contributor guide](https://jupyter.readthedocs.io/en/latest/contributor/content-contributor.html).
+Welcome! As a [Jupyter](https://jupyter.org) project, we follow the
+[Jupyter contributor guide](https://jupyter.readthedocs.io/en/latest/contributor/content-contributor.html).
 
-To develop binderhub, you can use a local installation of JupyterHub on minikube,
-and run binderhub on the host system.  Note that you will quickly hit your API limit
-on GitHub if you don't have a token.
+There are several different setups for developing BinderHub, depending on which
+parts of it you want to change: the [documentation](#documentation-changes),
+the [user interface](#user-innterface-changes), or the
+[kubernetes integration](#Kubernetes-integration-changes).
 
-## Installation
+
+## Documentation changes
```

Comment:
In the docs section, I'd mention that you use RST and link to a tutorial if people aren't familiar